### PR TITLE
[v1.0][W10-B3] HackEat

### PR DESCRIPTION
**Pull request by jaxony**
[Team Repo](https://github.com/CS2103Jan2018-W10-B3/main) 
For updates to master branch

@samzx @tohcheryl

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -800,15 +803,67 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 |Priority |As a ... |I want to ... |So that I can...
 |`* * *` |new user |see usage instructions |refer to instructions when I forget how to use the App
 
-|`* * *` |user |add a new person |
+|`* * *` |user |Use cmdline to add food that I like | I can get recommended meals that I will likely enjoy
```
Comment (tshradheya) [inline docs/DeveloperGuide.adoc]: `CLI` instead of `cmdline`

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -800,15 +803,67 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
...
+|`*` |user |See the rating of a restaurant |I know how reliable/how good the food is from that restaurant
+
+|`*` |user |Delete restaurants that I have saved | So that I can keep an updated list of restaurants I like
+
+|`*` |user with many foods in the address book |sort foods by name |locate a food easily
+
+|`*` |user |See a history of what I’ve written as a GitHub-style diagram |I can see something cool
 |=======================================================================
 
 _{More to be added}_
```
Comment (tshradheya) [inline docs/DeveloperGuide.adoc]: Remove this.

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -819,30 +874,73 @@ _{More to be added}_
 (For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
```
Comment (tshradheya) [inline docs/DeveloperGuide.adoc:1190]: Change instances of `AddressBook` to your App name

File: docs/DeveloperGuide.adoc [hunk 4]
```diff
@@ -819,30 +874,73 @@ _{More to be added}_
...
+
+* 1a. No food is sale
++
+[none]
+** 1a1. Hackeat informs user that nothing is on sale.
+** 1a2. User chooses to order other food.
++
+Use case resumes from Step 4.
 
 _{More to be added}_
```
Comment (tshradheya) [inline docs/DeveloperGuide.adoc]: Remove all these

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 5]
```diff
@@ -99,14 +99,6 @@ public void parse_compulsoryFieldMissing_failure() {
         assertParseFailure(parser, NAME_DESC_BOB + VALID_PHONE_BOB + EMAIL_DESC_BOB + ADDRESS_DESC_BOB,
                 expectedMessage);
 
-        // missing email prefix
-        assertParseFailure(parser, NAME_DESC_BOB + PHONE_DESC_BOB + VALID_EMAIL_BOB + ADDRESS_DESC_BOB,
```
Comment (tshradheya) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java]: How about writing a test to check that command doesnt fail if email isnt entered?

**Review by tshradheya — COMMENTED** [review]
Great job overall. Some points to note

To do:
- Add coveralls also for the project, will help you in knowing what tests to write
- Good Target User and User stories. Instead of having `users` everywhere, you can use adjectives to describe them for e.g. `**lazy users** so that they can order with a single command`
- Amazing job o NFR's and Use Cases
- After every version, update the PR description with summary of what everyone did.

**Comment by tshradheya** [comment]
Good work updating the UG, AboutUs and Readme.
Also great work refactoring 🎉 

More things can be added to UG especially whats coming in V2.0[in the table of content](which you may decide to implement or not).
Please refer to the announcement/book and do the necessary for next week.
Each member of the team should have 1 merged PR of their enhancement atleast as its V1.2 release(It doesnt need to be perfect code, but something that atleast works).

If you guys, want me to review your code for quality checks tag me in the pr you make to your team repo.

Note: Every weekend please update your PR title to new version name and give a summary of what each of you have done for the week

**Comment by samzx** [comment]
**Weekly Summary:**

Sam
- Added tests for last week's PR review (address and email fields being optional)
- Added Feature contribution in developer guide
- Created a Rating class as an extra property for Food (Minor v1.2)
- Created developer guide for `order` command with diagrams

**Comment by jaxony** [comment]
**Weekly Summary 21 Mar**
- implemented minor enhancement: `userconfig` (unmerged)
- updated developer guide (merged)
- updated user guide (merged)

**Comment by jaxony** [comment]
**Weekly Summary 14 Mar**
- individual: refactored AddressBook to HackEat
- team task: set up automated documentation publishing

**Comment by tshradheya** [comment]
Great job with V1.2. Keep going with the same pace :)

This review will only be for your V1.2 DG, I will give you a review for your code near the end [ you can tag me in your PR to the team if you want me to do code review at an early stage also]

Points to Note:

1. 
image

its `.PNG` instead of `.png`
I know this feature isn't already implemented, so when it is done, add more content to the DG for it.
Instead of telling what data structure is used, focus more on the architecture and how it works[the sequence]

2. 
image

This code is really unnecessary, only put code snippets of what is unique to this feature. If you want to show parsing then you can show it using Sequence diagram instead of this code snippet.
Good job on dividing into parts 🎉 
Maybe add a few UML diagrams?

3. 
image

DG should have more focus on `HOW` part and tell how the feature is implemented and its design in a top-down manner.
What kind of UML diagram is this? :P 
If you don't use standard annotation, then you need to mention the difference.


General Tip: 
- Add more diagrams
- Try to follow a similar style as `Undo-Redo section in DG`
- diagrams/examples in their description of how a feature is implemented
- The description of how it works could be better :)
- Elaborate more on Pros and Cons


Current documentation still has a lot of room for improvement (both user guide and developer guide). Please put more effort into it.

**Comment by jaxony** [comment]
# Week ending: Wednesday 4 April
## Individual contributions
- finished logic / basic UI of major feature (chat-like interaction)
- finished tests for above
- merged all code into `master`
- updated `UserGuide.adoc`

## Team contributions
- updated `README.adoc`: changed `Ui.png` from mock to screenshot
- fixed `README.adoc`: fixed `.jar` name and team repo url, removed old instructions
- project management: closed, reviewed, approved, merged PRs, closed milestone, created `v1.4` release with `.jar`
- fixed `.jar` filename in `build.gradle`
- collation: set up code collation `bash` script and merged collation source code into master

**Comment by samzx** [comment]
@jaxony can we close this pr 😛

**Comment by jaxony** [comment]
@samzx still blocked :(

**Pull request closed without merging**